go-controller/pkg/ovn/gateway.go

```diff
@@ -721,17 +643,33 @@ func (gw *GatewayManager) GatewayInit(
 			}
 		}
 	}
+	return nil
+}
 
+// updateGWRouterStaleNAT updates the SNAT rules on the gateway router that are created outside the GatewayManager.
+// Multiple handlers, like
+// - DefaultNetworkController.addLogicalPort
+// - DefaultNetworkController.updateNamespace
+// - EgressIPController.addExternalGWPodSNATOps
+// - EgressIPController.addPodEgressIPAssignment
+// - SecondaryLayer2NetworkController.buildUDNEgressSNAT
+// - SecondaryLayer3NetworkController.addUDNNodeSubnetEgressSNAT
+// use gateway config parameters to create SNAT rules on the gateway router, but some of them (not all) don't watch
+// gateway config changes and rely on the GatewayManager to update their SNAT rules.
+// Is it racy? Yes!
+func (gw *GatewayManager) updateGWRouterStaleNAT(externalIPs, oldExtIPs, gwLRPIPs []net.IP,
```

Comment:
still dont really like the name because stale NAT handling is also happening in updateGWRouterNAT->gw.cleanupStalePodSNATs

It looks like this code originates from:
https://github.com/ovn-kubernetes/ovn-kubernetes/commit/264697ddf710859e7e833b84b0b3aeeea949f323

and the purpose of it looks to detect if the node IP or the join IP changes on the GR. Maybe we can rename this function to "syncNATsForGRIPChange" ?

Reply:
updated name and comment